### DISCUSSION TITLE
RESTRICTIVE RLS POLICY not working

### DISCUSSION THREAD

**Discussion by TimDR-1356** [supabase/supabase#30671] [Questions] [answered]
I have to make a RESTRICTIVE policy because I want to use multiple once to act together. Now I keep getting no information back when I create a RESTRICTIVE policy, even if I just set it to true, which I find strange. see the picture below. I also have no different policy on the table. In the pictures below you can see that i only have that policy on the table and that there is data. But if i change rol to a authenticated user then it shows no data. Am I doing something wrong, or is this a bug? I also created a new project to see if it is related to the project, but I get the same thing there.

<img width="534" alt="Screenshot 2024-11-26 at 11 52 17" src="https://github.com/user-attachments/assets/e8b6b4be-48e6-4d43-82ff-8747b449e4aa">

<img width="339" alt="Screenshot 2024-11-26 at 11 53 05" src="https://github.com/user-attachments/assets/e783e2dd-a767-4177-9c46-ce1189e4821a">
<img width="884" alt="Screenshot 2024-11-26 at 11 53 53" src="https://github.com/user-attachments/assets/976131e4-e6d4-4607-a010-3a9652d03ad4">
<img width="722" alt="Screenshot 2024-11-26 at 11 53 37" src="https://github.com/user-attachments/assets/efafdcf2-4b8d-45c8-afaf-4faf2a18bb27">

**Comment by GaryAustin1** [comment, marked as answer]
In order to use a restrictive policy you first have to have a permissive policy.
https://www.postgresql.org/docs/current/sql-createpolicy.html
